src/system/SystemPacketBuffer.cpp

```diff
@@ -508,45 +508,71 @@ void PacketBuffer::AddRef()
 
 PacketBufferHandle PacketBufferHandle::New(size_t aAvailableSize, uint16_t aReservedSize)
 {
-    // Adding three 16-bit-int sized numbers together will never overflow
-    // assuming int is at least 32 bits.
+    // Ensure that SIZE_MAX is at least 32 bits
     static_assert(INT_MAX >= INT32_MAX, "int is not big enough");
+    static_assert(SIZE_MAX >= INT_MAX, "Our additions may not fit in size_t");
+
+    // Sanity check for kStructureSize and ensure that the constant sizes are not going to lead
+    // to overflow.
```

Comment:
please add some why here - what is the UINT16 that controls size of things (given that pbuf sizes are size_t)? When trying to read this, I am wondering if I set max sizes, I have `kMaxSizeWithoutReserve` being UINT16_MAX and then we reserve space for something ... so the entire pbuf size will be over UINT16_MAX ... is that ok? It seeems hard to follow how these size choices were made.

https://github.com/project-chip/connectedhomeip/blob/master/src/system/SystemPacketBuffer.h#L94 seems to describe the data layout however that does not immediately indicate why uint16 is important.

Reply:
Modified comment slightly.